src/Umbraco.Web.Common/Filters/OutgoingNoHyphenGuidFormatAttribute.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Buffers;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Umbraco.Core;
+using Umbraco.Web.Common.Formatters;
+
+namespace Umbraco.Web.Common.Filters
+{
+    public class OutgoingNoHyphenGuidFormatAttribute : TypeFilterAttribute
+    {
+        public OutgoingNoHyphenGuidFormatAttribute() : base(typeof(OutgoingNoHyphenGuidFormatFilter))
+        {
+            Order = -2000;
+        }
+
+        private class OutgoingNoHyphenGuidFormatFilter : IResultFilter
+        {
+            private readonly IOptions<MvcNewtonsoftJsonOptions> _mvcNewtonsoftJsonOptions;
+            private readonly ArrayPool<char> _arrayPool;
+            private readonly IOptions<MvcOptions> _options;
+
+            public OutgoingNoHyphenGuidFormatFilter(ArrayPool<char> arrayPool, IOptions<MvcOptions> options)
+            {
+                _arrayPool = arrayPool;
+                _options = options;
+            }
+            public void OnResultExecuted(ResultExecutedContext context)
+            {
+            }
+
+            public void OnResultExecuting(ResultExecutingContext context)
+            {
+                if (context.Result is ObjectResult objectResult)
+                {
+                    var serializerSettings = new JsonSerializerSettings();
+                    serializerSettings.Converters.Add(new GuidNoHyphenConverter());
+
+                    objectResult.Formatters.Clear();
+                    objectResult.Formatters.Add(new AngularJsonMediaTypeFormatter(serializerSettings, _arrayPool, _options.Value));
```

Comment:
@bergmania I don't understand why this and the JsonDateTimeFormatAttribute both clear and add the AngularJsonMediaTypeFormatter? That's is the responsibility of AngularJsonOnlyConfigurationAttribute. Is there a reason why this same logic is duplicated to these other 2 attributes?

Reply:
That's true.. The issue is if they don't it will be overridden.  I'm not sure what is the best way to "merge" this info when the formatters are set in different attributes.. Any suggestions? Basically we need to manipulate the JsonSerializerSettings of the existing formatters, if they are of type `NewtonsoftJsonOutputFormatter` 😕